src/cockpit/389-console/src/lib/plugins/dna.jsx

```diff
@@ -1249,9 +1250,11 @@ class DNA extends React.Component {
                             <div className="ds-margin-top-xlg">
                                 <DNASharedTable
                                     rows={sharedConfigRows}
+                                    key={sharedConfigRows}
```

Comment:
This is more of a general question, can we use an array as a key, my understanding is that sharedConfigRows can be modified by the user so it changes. Or am i not getting something ?

Reply:
So, the key can be used to tell the component that it needs to be re-rendered.  Without the key if sharedCofigRows changed it would not be reflected in the UI/component.  So when the key changes from the last render. then it tells the component to render from scratch again.

Arrays work as keys but objects do not.  So you will see places where I use a index counter as the key (tableKey) instead of an array or other variable object.

Reply:
> So, the key can be used to tell the component that it needs to be re-rendered. Without the key if sharedCofigRows changed it would not be reflected in the UI/component. So when the key changes from the last render. then it tells the component to render from scratch again.
> 
> Arrays work as keys but objects do not. So you will see places where I use a index counter as the key (tableKey) instead of an array or other variable object.

Ahh I see, that's handy. Thanks for the explanation.